### PR TITLE
Anonymous User fix for User Component

### PR DESCRIPTION
**Pull request by alexyoung23j**
## Link to Issue
Closes: #3840 

## Description of Changes
- Removes the <AnonymousUser /> component in favor of expanding <User /> to accept undefined users and correctly displaying them. This is required in some edge cases where an address is not the same base as the chain it is commented in. 

## Test Plan
- Tested locally. 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/user/user.tsx [hunk 1]
```diff
@@ -219,14 +226,14 @@ export const User = (props: UserAttrs) => {
...
-                !profile?.id ? (
-                  `${profile.address.slice(0, 8)}...${profile.address.slice(
-                    -5
-                  )}`
+              {!profile ? (
+                showAsDeleted ? (
+                  'Deleted'
                 ) : (
-                  addrShort
+                  'Anonymous'
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/components/user/user.tsx:233]: I think we want to show the addrShort UNLESS the address meets the conditions where we were showing AnonymousUsers, i.e.:
```
      comment.authorChain !== app.chain.id &&
      comment.authorChain !== app.chain.base
```

**Review by jnaviask — CHANGES_REQUESTED** [review]

File: packages/commonwealth/client/scripts/views/components/user/user.tsx (hunk 1 above)
Comment (alexyoung23j) [reply packages/commonwealth/client/scripts/views/components/user/user.tsx:233]: Yeah that is what's happening here; the entire ternary changed because it was written in a weird way so this isnt actually just swapping out "addrShort" for "Anonymous".

**Review by jnaviask — APPROVED** [review]
